community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/commands/predicates/Equivalent.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compiler.v3_0.commands.predicates
+
+import org.neo4j.cypher.internal.compiler.v3_0.{CRS, GeographicPoint}
+import org.neo4j.graphdb.{Node, Path, Relationship}
+
+import scala.collection.GenTraversableOnce
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ArrayBuffer
+
+// Class that calculates if two values are equivalent or not.
+// Does not handle NULL values - that must be handled outside!
+class Equivalent(protected val eagerizedValue: Any, val originalValue: Any) extends scala.Equals {
+  override def equals(in: Any): Boolean = {
+    val eagerOther = in match {
+      case s: Equivalent =>
+        if(originalValue.isInstanceOf[AnyRef] &&
+          s.originalValue.isInstanceOf[AnyRef] &&
+          (originalValue.asInstanceOf[AnyRef] eq s.originalValue.asInstanceOf[AnyRef])) return true
+        s.eagerizedValue
+      case x =>
+        if(originalValue.isInstanceOf[AnyRef] &&
+          x.isInstanceOf[AnyRef] &&
+          (originalValue.asInstanceOf[AnyRef] eq x.asInstanceOf[AnyRef])) return true
+        Equivalent.eager(x)
+    }
+
+    (eagerizedValue, eagerOther) match {
+      case (null, null) => true
```

Comment:
It says above that this class does not handle `null` values...? Which variables is that comment directed at?